lib/elixir/lib/application.ex

```diff
@@ -738,16 +738,18 @@ defmodule Application do
   """
   @spec delete_env(app, key, timeout: timeout, persistent: boolean) :: :ok
   def delete_env(app, key, opts \\ []) when is_atom(app) do
-    maybe_warn_on_app_env_key(key)
+    maybe_warn_on_app_env_key(app, key)
     :application.unset_env(app, key, opts)
   end
 
   # TODO: turn warning into error on Elixir v2.0
-  defp maybe_warn_on_app_env_key(key) when is_atom(key),
+  defp maybe_warn_on_app_env_key(_app, key) when is_atom(key),
     do: :ok
 
-  defp maybe_warn_on_app_env_key(key),
-    do: IO.warn("passing non-atom as application env key is deprecated (got: `#{inspect(key)}`)")
+  defp maybe_warn_on_app_env_key(app, key) do
+    message = "passing non-atom as application env key is deprecated (got: `#{inspect(key)}`)"
```

Comment:
We don't use `` ` `` in Elixir error messages. AFAIK Ecto is the only project that does it. :-)

```suggestion
    message = "passing non-atom as application env key is deprecated, got: #{inspect(key)}"
```

Reply:
<img width="1196" alt="image" src="https://user-images.githubusercontent.com/76071/77820608-d44e2680-70e3-11ea-8cb8-90530b513b94.png">